File: easy_1/word-count/js/word_count.js
class Phrase {
  constructor(inputPhrase) {
    this.phrase = inputPhrase;
  }

  wordCount() {
    return this.phrase
    .toLowerCase()
    .match(/\b[\w]+'?[\w]?\b/ig)
    .reduce( (wordCounts, word) => {
      wordCounts[word] ? wordCounts[word] += 1 : wordCounts[word] = 1;
      return wordCounts;
    }, {});
  }
}

export default Phrase;

File: easy_1/word-count/ruby/v2/word_count.rb
# word_count.rb

class Phrase
  def initialize(input_phrase)
    @phrase = input_phrase
  end

  def word_count
    phrase
    .downcase
    .scan(/\b[\w]+'?[\w]?\b/i)
    .reduce({}) do |count_dict, word|
      count_dict[word] ? count_dict[word] += 1 : count_dict[word] = 1
      count_dict
    end
  end

  protected
  attr_reader :phrase
end
